### ISSUE TITLE
Output `.zenodo.json`

### ISSUE THREAD

**Issue by hlmore** [Materials-Data-Science-and-Informatics/somesy#100]
Zenodo is a popular repository to assign code a DOI.  Code owners can either upload an archive of code and manually fill out the metadata fields, or use the [GitHub-Zenodo integration](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content) to automatically generate a DOI and landing page each time a new version of code is released.

When using the GitHub-Zenodo integration, if the GitHub repository contains a `.zenodo.json` file, Zenodo uses metadata from this file instead of requiring metadata to be entered manually.

More recently, Zenodo added support for `CITATION.cff`, and will use metadata from a `CITATION.cff` file if no `.zenodo.json` file is present.  However, `CITATION.cff` doesn't yet support all the fields that `.zenodo.json` does.

In particular, Zenodo supports a `grants` field which is crucial for scientific researchers required to report grant funding for their project outputs.  For European Commission grants, Zenodo allows researchers to only enter the grant number, which is extremely convenient for the many people receiving this funding.

It would be very useful if Somesy included an option to output a `.zenodo.json` file alongside the existing `CITATION.cff` and `codemeta.json` files, and supported more of Zenodo's metadata fields.

If Somesy did this, I would suggest that some of the most urgent Zenodo-specific fields to add would be:
- `contributors`
- `related_identifiers`
- `grants`
- `upload_type`
- 
Unfortunately just adding fields to the current `CITATION.cff` output by Somesy isn't sufficient, since cff doesn't (yet) support all of Zenodo's metadata fields.

Supported fields in:
- Zenodo:  https://developers.zenodo.org/#representation
- `CITATION.cff`:  https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#index
- Somesy:  https://materials-data-science-and-informatics.github.io/somesy/main/manual/#somesy-metadata-schemas

**Comment by daniel-mohr** [comment]
Yes, the output of `.zenodo.json` would really be a useful improvement as there are many more fields available.

But I don't think this only makes sense for GitHub. There is already an integration for GitHub that also supports CITATION.cff.

However, this integration is limited to GitHub and lacks comparable functionality for other open source platforms such as GitLab or Gitea.

That's why I created the [deploy2zenodo](https://gitlab.com/deploy2zenodo/deploy2zenodo) tool.

When deploying deploy2zenodo to zendoo, I even supplement the CITATION.cff in the first step with current information that is only available in a CI pipeline, for example:

```sh
TMPCITATION=$(mktemp)
yq --yaml-output ".url = \"$CI_PROJECT_URL/-/releases/$TAG\" | .commit = \"$TAG_COMMIT_SHA\" | .version = \"$TAG\" | .\"date-released\" = \"$TAG_COMMIT_TIMESTAMP\" | .identifiers[0].description = \"The DOI of this release.\" | .identifiers[0].value = \"$DOI\" | .identifiers[0].type = \"doi\" | .doi = \"$DOI\"" "$CITATION" | tee "$TMPCITATION"
mv "$TMPCITATION" "$CITATION"
cffconvert -i "$CITATION" --validate
```


In a later step, I add static information, for example:

```sh
cffconvert -i CITATION.cff -f zenodo | \
jq -c --argjson mixin \
"$(jq .metadata.related_identifiers "zenodo-add-related_identifiers.json")" \
--rawfile README "$tmpreadme" \
"{\"metadata\": .} | .metadata.description = \$README |
          .metadata.url = \"$tagurl\" |
          .metadata.related_identifiers += [
            {\"relation\": \"isDerivedFrom\",
             \"identifier\": \"$projecturl\",
             \"scheme\": \"url\",
             \"resource_type\": \"software\"},
            {\"relation\": \"isSupplementTo\",
             \"identifier\": \"$tagurl\",
             \"scheme\": \"url\",
             \"resource_type\": \"software\"},
             {\"relation\":\"IsNewVersionOf\",
             \"identifier\":\"$LATESTDOI\",
             \"scheme\":\"doi\",
             \"resource_type\":\"$LATESTUPLOADTYPE\"}] |
          .metadata.related_identifiers += \$mixin" | \
cat "zenodo-merge.json" - | jq -c -s '.[0] * .[1]' | \
tee "$DEPLOY2ZENODO_JSON" | jq -C .
```

This means that in addition to the output of `.zenodo.json`, the dynamic input would also be an improvement.
